frontend/test/e2e/query_builder/query_builder.spec.js

```diff
@@ -0,0 +1,155 @@
+    waitForElementRemoved,
+    findElement,
+    waitForElementAndClick,
+    waitForElementAndSendKeys,
+    waitForUrl,
+    screenshot,
+    loginMetabase,
+    describeE2E,
+    ensureLoggedIn
+} from "../support/utils";
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
+
+describeE2E("query_builder", () => {
+    beforeEach(async () => {
+        await ensureLoggedIn(server, driver, "bob@metabase.com", "12341234");
+    });
+
+    describe("tables", () => {
+        it("should allow users to create pivot tables", async () => {
+            await driver.get(`${server.host}/q`);
+
+            await screenshot(driver, "screenshots/qb-initial.png");
+
+            await waitForElementAndClick(driver, "#TablePicker .List-item:first-child>a");
+
+            await waitForElementAndClick(driver, "#Query-section-aggregation");
+            await waitForElementAndClick(driver, "#AggregationPopover .List-item:nth-child(2)>a");
+
+            await waitForElementAndClick(driver, ".Query-section.Query-section-breakout #BreakoutWidget");
+            await waitForElementAndClick(driver, "#BreakoutPopover .List-section:nth-child(3) .List-section-header");
+            await waitForElementAndClick(driver, "#BreakoutPopover .List-item:nth-child(12)>a");
+
+            await waitForElementAndClick(driver, ".Query-section.Query-section-breakout #BreakoutWidget .AddButton");
+            await waitForElementAndClick(driver, "#BreakoutPopover .List-item:first-child .Field-extra>a");
+            await waitForElementAndClick(driver, "#TimeGroupingPopover .List-item:nth-child(4)>a");
+
+            await waitForElementAndClick(driver, ".Button.RunButton");
+
+            await waitForElementRemoved(driver, ".Loading", 20000);
+            await screenshot(driver, "screenshots/qb-pivot-table.png");
+
+            // save question
+            await waitForElementAndClick(driver, ".Header-buttonSection:first-child");
+            await waitForElementAndSendKeys(driver, "#SaveQuestionModal input[name='name']", 'Pivot Table');
+            await waitForElementAndClick(driver, "#SaveQuestionModal .Button.Button--primary");
+
+            // add to new dashboard
+            await waitForElementAndClick(driver, "#QuestionSavedModal .Button.Button--primary");
+            await waitForElementAndSendKeys(driver, "#CreateDashboardModal input[name='name']", 'Main Dashboard');
+            await waitForElementAndClick(driver, "#CreateDashboardModal .Button.Button--primary");
+
+            // save dashboard
+            await waitForElementAndClick(driver, ".EditHeader .Button.Button--primary");
+            await waitForElementRemoved(driver, ".EditHeader");
+        });
+    });
+
+    describe("charts", () => {
+        xit("should allow users to create line charts", async () => {
+            await driver.get(`${server.host}/q`);
+
+            // select orders table
+            await waitForElementAndClick(driver, "#TablePicker .List-item:first-child>a");
```

Comment:
hmm. this feels fragile (specifically that the orders table is the first child of the table picker.)